apm-agent-plugins/apm-micrometer-plugin/src/main/java/co/elastic/apm/agent/micrometer/MicrometerInstrumentation.java

```diff
@@ -22,38 +22,27 @@
 import co.elastic.apm.agent.impl.GlobalTracer;
 import io.micrometer.core.instrument.MeterRegistry;
 import net.bytebuddy.asm.Advice;
-import net.bytebuddy.description.NamedElement;
 import net.bytebuddy.description.method.MethodDescription;
 import net.bytebuddy.description.type.TypeDescription;
 import net.bytebuddy.matcher.ElementMatcher;
 
 import java.util.Collection;
 import java.util.Collections;
 
-import static net.bytebuddy.matcher.ElementMatchers.hasSuperType;
-import static net.bytebuddy.matcher.ElementMatchers.isPublic;
-import static net.bytebuddy.matcher.ElementMatchers.isStatic;
-import static net.bytebuddy.matcher.ElementMatchers.nameContains;
 import static net.bytebuddy.matcher.ElementMatchers.named;
-import static net.bytebuddy.matcher.ElementMatchers.not;
 
 public class MicrometerInstrumentation extends TracerAwareInstrumentation {
 
     private static final MicrometerMetricsReporter reporter = new MicrometerMetricsReporter(GlobalTracer.requireTracerImpl());
 
-    @Override
-    public ElementMatcher<? super NamedElement> getTypeMatcherPreFilter() {
-        return nameContains("Meter").or(nameContains("Registry"));
-    }
-
     @Override
     public ElementMatcher<? super TypeDescription> getTypeMatcher() {
-        return hasSuperType(named("io.micrometer.core.instrument.MeterRegistry"));
+        return named("io.micrometer.core.instrument.MeterRegistry");
     }
 
     @Override
     public ElementMatcher<? super MethodDescription> getMethodMatcher() {
-        return isPublic().and(not(isStatic()));
+        return named("registerMeterIfNecessary");
```

Comment:
As this is a private method, we should add micrometer to the allow list in [`dependabot.yml`](https://github.com/elastic/apm-agent-java/blob/master/.github/dependabot.yml) and always test against the latest version. To ensure compatibility with older versions, let's add a `VersionsIT`.

Reply:
Agreed, a `VersionsIT` is required for the instrumentation test. 
I already know that upgrading the default version to the latest will require adaptation of the reporter test, so we'll first verify this solves the problem.

Reply:
Added